util/validate/index.mjs

```diff
@@ -45,78 +45,89 @@ const run = ({
-    .through(getSchemaContent)
-    .tap(addSchema)
-    .collect()
-    // Validate all of the JSON-Schemas
-    .flatMap(_ => h(getAllSchemas()))
-    .flatMap(x => {return h(validateJsonSchema(x))})
-    .tap(result => {
-      if (result.valid) {
-        logSuccess(`${result.title} is valid`)
-      }
-      else {
-        console.error(`\nERROR: ${result.title} failed validation`)
-        errors ++
-      }
-    })
-    .collect()
-    // Switch to OpenRPC modules
-    .flatMap(_ => recursiveFileDirectoryList(externalFolder))
-    .through(getSchemaContent)
-    .tap(addSchema)
-    .flatMap(x => {return h(validateJsonSchema(x))})
-    .tap(result => {
-      if (result) {
-        logSuccess(`${result.title} is valid`)
-      }
-      else {
-        console.error(`\nERROR: ${result.title} failed validation`)
-        errors ++
-      }
-    })
-    .collect()
-    // Switch to OpenRPC modules
-    .flatMap(_ => recursiveFileDirectoryList(modulesFolder))
-    .through(getModuleContent)
-    .flatMap(x => {return h(validateOpenRpc(x))})
-    .tap(result => {
-      if (result.valid) {
-        logSuccess(`${result.title} is valid`)
-      }
-      else {
-        console.error(`\nERROR: ${result.title} failed validation`)
-        errors ++
-      }
-    })
-    .collect()
-    // Switch to merged OpenRPC file
-    .flatMap(_ => recursiveFileDirectoryList(distFolder))
-    .through(getModuleContent)
-    .flatMap(x => {return h(validateOpenRpc(x))})
-    .tap(result => {
-      if (result.valid) {
-        logSuccess(`${result.title} is valid`)
-      }
-      else {
-        console.error(`\nERROR: ${result.title} failed validation`)
-        errors ++
-      }
-    })
-    .done(() => {
-      if (errors > 0) {
-        console.error(`\nValidation failed with errors in ${errors} files`)
-        process.exit(1000)
-      }
-    })
+  if (path.extname(srcFolderArg) === '.json') {
+    h.of(srcFolderArg)
+      .through(getSchemaContent)
+      .flatMap(x => {return h(validateJsonSchema(x))})
+      .tap(result => {
+        if (result.valid) {
+          logSuccess(`OpenRPC: ${result.title} is valid`)
+        }
+        else {
+          console.error(`\nERROR: ${result.title} failed validation`)
+          errors ++
+        }
+      })
+      .done(() => {
+        if (errors > 0) {
+          console.error(`\nValidation failed with errors in ${errors} files`)
+          process.exit(1000)
+        }
+      })
+  }
+  else {
+    // Load all of the JSON-Schemas
+    recursiveFileDirectoryList(sharedSchemasFolder).flatFilter(isFile)
+      .through(getSchemaContent)
+      .tap(addSchema)
+      .collect()
+      .flatMap(_ => recursiveFileDirectoryList(schemasFolder).flatFilter(isFile))
+      .through(getSchemaContent)
+      .tap(addSchema)
+      .collect()
+      // Validate all of the JSON-Schemas
+      .flatMap(_ => h(getAllSchemas()))
+      .flatMap(x => {return h(validateJsonSchema(x))})
```

Comment:
Sorry for the annoying "style" comment, but this can be a one-liner like the other simple `.flatMap` s in here:
```js
.flatMap(x => h(validateJsonSchema(x))
```

Reply:
not annoying at all!